### PR TITLE
Custom Reports update TS issues

### PR DESCRIPTION
**Pull request by carkom**
Fixing typescript issues with firstDayOfWeek. Also fixes bug with TableGraph report card.

### CODE REVIEW

File: packages/desktop-client/src/components/reports/reports/GetCardData.tsx [hunk 1]
```diff
@@ -31,19 +33,67 @@ function ErrorFallback() {
...
   accounts: AccountEntity[];
   categories: { list: CategoryEntity[]; grouped: CategoryGroupEntity[] };
   earliestTransaction: string;
+  firstDayOfWeekIdx?: LocalPrefs['firstDayOfWeekIdx'];
 }) {
+  const fromDate = convertFromDate(report.interval);
+  const rangeInclusive = convertRangeInclusive(report.interval);
+
+  let dateStart;
+  let dateEnd;
```
Comment (twk3) [inline packages/desktop-client/src/components/reports/reports/GetCardData.tsx:81]: dateStart and dateEnd are redefined as constants when the report doesn't have a static date a little below. Likely using a different variable name here will help reduce confusion about them.

Also, do we need this section to be below the call to getLiveRange and use the outcomes of that?

File: packages/loot-core/src/shared/months.ts [hunk 2]
```diff
@@ -437,3 +441,26 @@ export const getShortYearRegex = memoizeOne((format: string) => {
     .replace(/y+/g, '\\d{2}');
   return new RegExp('^' + regex + '$');
 });
+
+function convertWeekDay(
+  firstDayOfWeekIdx: LocalPrefs['firstDayOfWeekIdx'],
+): 0 | 1 | 2 | 3 | 4 | 5 | 6 {
+  switch (firstDayOfWeekIdx) {
```
Comment (twk3) [inline packages/loot-core/src/shared/months.ts:448]: Can't we use parseInt here instead?

File: packages/desktop-client/src/components/reports/reports/GetCardData.tsx (hunk 1 above)
Comment (carkom) [reply packages/desktop-client/src/components/reports/reports/GetCardData.tsx:81]: Yes and yes. Done and done.

**Comment by twk3** [comment]
@carkom I'm running into some inconsistent behaviour with the firstDayOfWeek Pref still here. Where the saved report (and thus the card) doesn't reflect what is show via changing the intervals.

To repro what I am seeing:

- Create a new test budget
- Enable custom reports
- Set day of week to Wednesday.
- Save a bar graph. Interval: Weekly, Range: Last week
Screenshot from 2024-04-20 10-58-37
- Go back and view the card. (And even re-entering the report, and you see just the data as if first day of week was set to 0)
Screenshot from 2024-04-20 10-59-08

**Comment by carkom** [comment]
> @carkom I'm running into some inconsistent behaviour with the firstDayOfWeek Pref still here. Where the saved report (and thus the card) doesn't reflect what is show via changing the intervals.

Found the issue and fixed it!

**Review by twk3 — APPROVED** [review]
@carkom thanks, I confirmed that's fixed!

This looks ready to merge :rocket:

**Pull request merged**